text/0045-transaction_fee_minting.md

```diff
@@ -0,0 +1,232 @@
+- The deposit-based token (D) does not pay a fee.
+- The fee-based token (B) pays the standard fee.
+
+```
+- Base Fee: 0.001 HTR
+- Total Fee: 0.001 HTR (no extra charge for Token D)
+```
+
+#### Transaction Interacting with a Nano Contract
+Description: A transfer involving a nano contract, which may require additional execution on the network.
+
+Fee Calculation:
+- The standard transaction base fee still applies.
+- Since nano contract execution involves additional computation, an extra fee may be charged based on complexity.
+- Execution Fee (depends on contract complexity): Example, 0.002 HTR.
+
+```	
+- Base Fee: 0.001 HTR
+- Total Fee: 0.003 HTR (0.001 base + 0.002 contract execution)
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+TODO
+
+# Drawbacks of the Hybrid Fee Model
+[drawbacks]: #drawbacks
+
+While the hybrid fee model balances incentives and network sustainability, it also introduces potential drawbacks that must be considered. Below are key concerns and challenges:
+
+### 1. Token Inflation and Supply Growth
+
+**Issue:** Since there is no burning mechanism tied to fee-based token transfers, the total supply of tokens may continuously increase without a natural deflationary counterbalance.
+
+**Impact:** Over time, token supply inflation can reduce the value of individual tokens, especially if demand does not grow proportionally.
+
+**Potential Mitigation:**
+  - Introduce a **burn mechanism** where a percentage of the transaction fee is permanently removed from circulation.
+  - Implement an **optional melt requirement** where token issuers must periodically remove supply from circulation.
+
+---
+
+### 2. Potential for Low-Cost Network Congestion
+
+**Issue:** If fees are too low, bad actors could create a high volume of low-value transactions, leading to congestion.
+
+**Impact:**
+  - Increased load on network validators and miners.
+  - Slower transaction processing times during peak usage.
+
+**Potential Mitigation:**
+  - Introduce **dynamic fee adjustments** where fees scale based on network congestion.
+  - Set **minimum transaction fees** to prevent microtransactions from overloading the system.
+
+---
+
+### 3. Complexity in Multi-Token Transactions
+
+**Issue:** Transactions involving multiple tokens introduce a more complex fee structure, requiring additional computation to determine the appropriate charge.
+
+**Impact:**
+  - Increased processing overhead for nodes.
+  - Potential user confusion when sending transactions with different token types.
+
+**Potential Mitigation:**
+  - Improve **wallet UX** to clearly display estimated fees before confirming transactions.
+  - Optimize **fee calculation logic** to minimize the processing impact.
+
+---
+
+### 4. Miner Incentive Model May Need Adjustments
+
+**Issue:** Since miners are currently incentivized through TX mining and HTR block rewards, a shift to fee-based transactions could impact their revenue model.
```

Comment:
Tx mining is not part of the miners incentive scheme. Additionally, I can't see how the current proposal would impact miners since all previous mechanisms would still be in place.

Reply:
Ok.